### ISSUE TITLE
No envia por mail las respuestas

### ISSUE THREAD

**Issue by jordiribes** [opensupports/opensupports#641]
Hola, 

Primero felicitaros por la aplicación.

Comentar que si envias un ticket nuevo o lo cierras el sistema envía un email al creador del ticket, pero si se le responde el ticket el usuario no recibe la notificación de que se ha respondido ese ticket.

Tengo configurado correctamente el SMPT ya que se envía las notificaciones de creación y borrado al creador del ticket, pero no recibe la notificación de las respuestas.

Gracias !!!!

**Comment by mazierovictor** [comment]
Hello, I also have the same problem, the smtp test is all right, but the system does not send any email to either the administrator or the user. Tried to go back to version 4.4.0 also has the same problem.

**Comment by briciocadena** [comment]
I have the same problem, smtp and imap test are OK, system create the ticket by mail, system send ticket status when close or create it, but don't send mail with the responses user doesn't receive notification mail.

Do you have any updates for this issue?

**Comment by MenAtWeb** [comment]
I have the same issue. The system does not notify the senders when their tickets have been responded.

**Comment by briciocadena** [comment]
I try access trough HTTP not HTTPS and it's working, could you try and validate? I'm search the way to work on HTTPS protocol, if you have more information please respond this thread.

**Comment by lapega** [comment]
Can confirm that using HTTPS the notifications were sent.

**Comment by uneasymusic** [comment]
so which one now ? it works when using http? or https?, I don't have https on my server and it doesn't work. 
so, which notifications should be sent then ? I only get initial ticket creation ticket back to the creator.

**Comment by lapega** [comment]
The notifications are sent only at using HTTPS.

**Comment by uneasymusic** [comment]
except they are not ;) I am not using https and getting notifications. I just don't think I get all notifications, but I  checked some code and looks like not all notifications are covered.

**Comment by peterbra** [comment]
For HTTPS - I can confirm that system is NOT sending email notifications to end-user when ticket is replied by staff member. Can somebody give a hint what's wrong ? ...I have both POP and IMAP working OK and when user sends email to "support@mydomain.com" it does open a ticket, but when staff reply notification to user isn't sent...

**Comment by uneasymusic** [comment]
Look into recent issues, I found the bug when system doesn't sent email to user if ticket doesn't have assign staff. Even if staff respond to ticket.
I also fixed that in the api code. I added extra emails notifications to user when priority changes, I am planning to add email notifications to assign staff only when it's getting assigned. 
Also when user replys to ticket and there is no staff assigned, nobody gets email. So I fixed that, and whole group gets email if not single staff assigned.

**Comment by peterbra** [comment]
Yes, this is TRUE - when staff member is assigned - it sends out notification, but now if users reply back to email - it doesn't update the ticket with content....

**Comment by romerocox** [comment]
Any update on this? I don't get any emails even with HTTPS. 

![image](https://user-images.githubusercontent.com/29244464/71681760-8f705900-2d5b-11ea-8060-964b5e8d54f7.png)

**Comment by uneasymusic** [comment]
I didn't try with HTTPS but that should have an affect on anything. The only problem I had, I had SMTP without authentication and or encryption, and that requires changing mailer option in the code. I don't know if that's the case but,
in api/libs/MailSender.php Mailer instance looks like that:
```php
    private function getMailerInstance() {
        if(!($this->mailerInstance instanceof PHPMailer)) {
            $this->mailerInstance = new PHPMailer();

            $this->mailerInstance->From = $this->mailOptions['from'];
            $this->mailerInstance->FromName = $this->mailOptions['fromName'];
            $this->mailerInstance->CharSet = 'UTF-8';

            $this->mailerInstance->isSMTP();
            $this->mailerInstance->SMTPAuth = false;
            $this->mailerInstance->Host = $this->mailOptions['smtp-host'];
            $this->mailerInstance->Username = $this->mailOptions['smtp-user'];
            $this->mailerInstance->Password = $this->mailOptions['smtp-pass'];
            $this->mailerInstance->Timeout = 3500;
            $this->mailerInstance->SMTPOptions = [
                'ssl' => [
                    'verify_peer' => false,
                    'verify_peer_name' => false,
                    'allow_self_signed' => true
                ]
            ];
        }

        return $this->mailerInstance;
    }
```

**Comment by aedwards86** [comment]
i solved this issue by commenting out a few lines of /api/controllers/ticket/comment.php

from about line 95, note the last but one line is NOT commented out. this sends the email to the ticket author if private is not ticked.

`     // if($isAuthor && $this->ticket->owner) {
       //     $this->sendMail([
       //         'email' => $this->ticket->owner->email,
       //         'name' => $this->ticket->owner->name,
       //         'staff' => true
       //     ]);
       // } else if($isOwner) {
         !Controller::request('private') ? $this->sendMail($ticketAuthor) : null;
       // }`

**Comment by JoelBonetR** [comment]
> i solved this issue by commenting out a few lines of /api/controllers/ticket/comment.php
> 
> from about line 95, note the last but one line is NOT commented out. this sends the email to the ticket author if private is not ticked.
> 
> ```
> // if($isAuthor && $this->ticket->owner) {
>        //     $this->sendMail([
>        //         'email' => $this->ticket->owner->email,
>        //         'name' => $this->ticket->owner->name,
>        //         'staff' => true
>        //     ]);
>        // } else if($isOwner) {
>          !Controller::request('private') ? $this->sendMail($ticketAuthor) : null;
>        // }
> ```

Is this safe to use as valid workaround? Does this validation lines commented could make a missfunction of the system and send the email to any user or something?
